### PR TITLE
cleanup warnings throughout codebase

### PR DESCRIPTION
**Pull request by yaahc**

### CODE REVIEW

**Comment by yaahc** [comment]
The code generated by `#[derive(Arbitrary)]` is triggering the `clippy::unit_arg` lint, I'll open an upstream issue / PR to get that cleaned up so we can remove these allows

**Comment by yaahc** [comment]
I had to mess with the match statement in zebra-network/src/protocol/external/codec.rs a bunch which made it more verbose, but I was getting weird compiler errors that were probably entirely from `rust-analyzer` and not indicative of a real problem. It seems like the `use Message::*` was name collisions with other types that were already in scope which was just making rust-analyzer fall over. Either that or rust analyzer exports some special setting that more comprehensively tests our code when it runs our code, but messing with the body of the code seemed to introduce compiler errors so im assuming it was all compiling fine.

File: zebra-network/src/peer/error.rs [hunk 1]
```diff
@@ -37,14 +37,6 @@ pub enum PeerError {
...
     #[error("Remote peer sent an unsupported message type.")]
     UnsupportedMessage,
-    /// We got a `Reject` message. This does not necessarily mean that
-    /// the peer connection is in a bad state, but for the time being
-    /// we are considering it a PeerError.
-    // TODO: Create a different error type (more at the application
-    // level than network/connection level) that will include the
-    // appropriate error when a `Reject` message is received.
-    #[error("Received a Reject message")]
-    Rejected,
```
Comment (dconnolly) [inline zebra-network/src/peer/error.rs]: I know this isn't being used yet, but until we have its replacement, I would like to keep it here

**Review by dconnolly — COMMENTED** [review]
😍

File: zebra-network/src/peer/error.rs (hunk 1 above)
Comment (yaahc) [reply zebra-network/src/peer/error.rs]: Sounds good

**Review by dconnolly — APPROVED** [review]
🚀

**Pull request merged**